### ISSUE TITLE
[FEATURE] Ability to define cropping for images in EditorInterface.yaml

### ISSUE THREAD

**Issue by secured** [FriendsOfTYPO3/content-blocks#41]
You can provide your own croppings for CTypes via TCA/Overrides.
It would be great if these could also be defined in the YAML file "EditorInterface.yaml".

The advantage would be that the configuration required for the image elements is also supplied with the "Content Blocks".

**Comment by nhovratov** [comment]
Thank you, we will think about this and provide a possibility to easily integrate this with Content Blocks.

**Comment by nhovratov** [comment]
For reference: https://docs.typo3.org/m/typo3/reference-tca/main/en-us/ColumnsConfig/Type/ImageManipulation/Properties/CropVariants.html

So normally to define croppings per element you would do this in TCA

```
'config' => [
   'overrideChildTca' => [
      'columns' => [
         'crop' => [
            'config' => [
               'cropVariants' => [
                  'mobile' => [
                     'title' => 'LLL:EXT:ext_key/Resources/Private/Language/locallang.xlf:imageManipulation.mobile',
                     'cropArea' => [
                        'x' => 0.1,
                        'y' => 0.1,
                        'width' => 0.8,
                        'height' => 0.8,
                     ],
                  ],
               ],
            ],
         ],
      ],
   ],
]
```
this could be simplified in the EditorInterface.yaml to this, as this is such a common task. In fact, this could even be added to Core as a feature.

```
- identifier: my_image
  type: File
  allowed: common-image-types
  cropVariants:
    mobile:
      allowedAspectRatios:
       4:3:
         value: 1.33333

```

**Comment by nhovratov** [comment]
It's implemented! Will be release with version 0.5.0 stay tuned, date is not planned yet.
Documentation can be found here: https://docs.typo3.org/p/contentblocks/content-blocks/main/en-us/YamlReference/FieldTypes/File/Index.html#confval-cropVariants
